template/cordova/lib/build.js

```diff
@@ -67,8 +67,15 @@ module.exports.run = function run (buildOptions) {
         }
 
         events.emit('log', 'Found VSINSTALLDIR environment variable. Attempting to build project using that version of MSBuild');
+        var msBuildPath = process.env.VSINSTALLDIR + "\\MSBuild\\15.0\\Bin";
+        fs.statSync(msBuildPath, function(err, stats) {
+            if (err && err.code === 'ENOENT') {
+                events.emit('log', 'Cannot find MSBuild path under VSINSTALLDIR. The following path does not exist: ' + msBuildPath);
```

Comment:
Since this will be the normal scenario for previous versions of VS (`VSINSTALLDIR` is set but there is no `msbuild` folder), I think we should avoid outputting too much noise (particularly a message that reads like there might be an error).

Basically, I think we should reverse the logic of this method - up front, if we find `VSINSTALLDIR` set, *and* `msBuildPath` exists, we should use that and output a single message to that effect. Otherwise (or if our attempt to read msbuild info from that folder fails), call `findAllAvailableVersions()` (this should also result in a much simpler method with only a single call to `findAllAvailableVersions()`).

That said, as I mentioned in the email thread I think it would be even *better* (and would also fix `check_reqs`) to simply move this logic into `findAllAvailableVersions()`.